### ISSUE TITLE
Training becomes slower and slower after epochs and results get deteriorated

### ISSUE THREAD

**Issue by C2H5OHlife** [ClubAI/MonoDepth-PyTorch#10]
Hello, I am strictly using the parameters according to advice from [here](https://github.com/ClubAI/MonoDepth-PyTorch/issues/1#issuecomment-400844411), but the result is getting worse along training process.
![figure_1](https://user-images.githubusercontent.com/9017429/47069599-e6a31000-d221-11e8-9bc0-ca004ccae485.png)

Another problem is the training process becomes slower (and the loss gets greater of course):
`Epoch: 1 train_loss: 1.8654187655635037 val_loss: 1.3897445498384196 time: 985.607 s
Model_saved`
`Epoch: 2 train_loss: 1.7081988794441578 val_loss: 1.299928411009348 time: 2059.05 s
Model_saved`
`Epoch: 3 train_loss: 1.6670600274022838 val_loss: 1.2872194408765298 time: 3121.77 s
Model_saved`
`Epoch: 4 train_loss: 1.6339830119013636 val_loss: 1.320097159008084 time: 4135.433 s`
`Epoch: 5 train_loss: 1.6094302785873893 val_loss: 1.505771610942589 time: 5179.27 s`

**Comment by voeykovroman** [comment]
Hello @C2H5OHlife! As we mentioned in another issue, we updated model: fixed several bugs in loss, changed bilinear sampler and so on and now we are training a new model. Therefore parameters described in README were used for the old model. Now we are trying to select new parameters for the new model and when it's trained we'll update provided model and description of training process. Meanwhile you can use old pretrained mode as mentioned [here](https://github.com/ClubAI/MonoDepth-PyTorch/issues/9#issuecomment-430435548) but keep in mind that it was far from perfect. Good luck!

**Comment by C2H5OHlife** [comment]
@voeykovroman Thanks for your reply! I've tried your old version, but the situation seemed remaining the same. This time epoch 0 took 4169.978s while epoch 9907.543s. And there're still some weird results coming out like below. (epoch 5 batch 850)

![figure_1-4](https://user-images.githubusercontent.com/9017429/47125688-7777fb00-d2b7-11e8-8def-5c6b44ef7a4a.png)

Is this just me or did you encounter this phenomenon when you're training?
Many thanks to you!

**Comment by voeykovroman** [comment]
@C2H5OHlife First of all, you can use the old model without training because it's already trained. Testing on new data also described in demo notebook. Of course until you need this model for some other task. 
What about slowdown it's pretty weird. We didn't encounter such problem (personally I didn't encounter such problem at all with any other models). Moreover this code was tested on at least 3 different machines so highly likely it works fine but it's true that training depth estimation model is quite painful and loss may not falling for significant amount of epochs.
So are you sure that all paths to training data are correct and data stored in according with the structure described in README? If it's not the issue I assume there may be some problems with software (version of Linux, PyTorch, CUDA and so on). May be you should check if everything work fine on other tasks, for example on some PyTorch tutorial.

**Comment by C2H5OHlife** [comment]
@voeykovroman thanks for your help!! I'm checking what's wrong with my machine now. Here are two more questions:
How much time did you spend for training your model?
And how do you select train and val set from the whole kitti dataset?

**Comment by NikolasEnt** [comment]
@C2H5OHlife
We trained the provided model for 100 epochs, each takes about 45 min on a single gtx 1080 ti gpu.
Now, we use the following subsets for validation:
```
2011_09_26_drive_0002_sync  2011_09_29_drive_0071_sync
2011_09_26_drive_0014_sync  2011_09_30_drive_0033_sync
2011_09_26_drive_0020_sync  2011_10_03_drive_0042_sync
2011_09_26_drive_0079_sync
```
However, the provided model was trained on the whole dataset without validation.

**Comment by NikolasEnt** [comment]
@C2H5OHlife
We manage to recreate your weird result images and updated the code in the master branch, fixed several bugs and provided a new pretrained model along with the updated jupyter notebook. You may try it.
However, we have no ideas for now why training time increases in your case.
